### DISCUSSION TITLE
What is "batch size" when using `getFullList()`

### DISCUSSION THREAD

**Discussion by babakfp** [pocketbase/pocketbase#1788] [Q&A] [answered]
Hi
Is it some kind of a limit? So it doesn't pull all the data, because it may take a long time or something like that?

```js
// you can also fetch all records at once via getFullList
const records = await pb.collection('events').getFullList(200 /* batch size */, {
    sort: '-created',
});
```

**Comment by ghost** [comment]
Batch size is how many items are selected and sent back in the response. So if batch size is `200` then `200` records will be returned.

You can use [filters](https://pocketbase.io/docs/api-rules-and-filters/#filters-syntax) to filter your responses.

**Comment by ganigeorgiev** [comment, marked as answer]
The batch size argument specifies how many items per requests to fetch at once **until** all items are fetched. So for example, if you have 300 items, `getFullList(200)` will make under the hood 2 consequent `getList()` requests with perPage=200 (the first one will resolve to 200 and the second one to 100 records). The final resolved Promise of `getFullList` will be always all items (aka. 300).
